external-crates/move/crates/move-compiler/src/typing/translate.rs

```diff
@@ -3102,25 +3105,47 @@ fn process_exp_dotted(
                 base_type,
                 accessors,
                 warn_on_constant: true,
+                for_autocomplete: false,
             }
         }
         N::ExpDotted_::Dot(ndot, field) => {
             let mut inner = process_exp_dotted(context, Some("dot access"), *ndot);
+            if inner.for_autocomplete {
+                return inner;
+            }
             let inner_ty = inner.last_type();
             let field_type = resolve_field(context, dloc, inner_ty, &field);
+            debug_print!(context.debug.autocomplete_resolution, ("field type" => field_type));
             inner.loc = dloc;
-            inner
-                .accessors
-                .push(ExpDottedAccess::Field(field, field_type));
+            if matches!(&field_type.value, Type_::UnresolvedError) && context.env.ide_mode() {
+                // If we failed to resolve a field, mark this as for_autocomplete to provide
+                // suggestions at the error location.
+                inner.for_autocomplete = true;
+            } else {
+                inner
+                    .accessors
+                    .push(ExpDottedAccess::Field(field, field_type));
+            }
             inner
         }
-        N::ExpDotted_::DotUnresolved(loc, ndot) => {
+        N::ExpDotted_::DotAutocomplete(_loc, ndot) => {
             let mut inner = process_exp_dotted(context, Some("dot access"), *ndot);
-            inner.accessors.push(ExpDottedAccess::UnresolvedError(loc));
+            if inner.for_autocomplete {
```

Comment:
This seems okay but makes me feel a bit uneasy. I wonder if it is worth just writing a separate `process_exp_dotted` for autocomplete

Reply:
Good point. I split out the functionality into shared bits, and feature gated the entire call into autocomplete for `ide_mdoe()`.